src/multivariate/product.jl

```diff
@@ -60,3 +62,17 @@ maximum(d::Product) = map(maximum, d.v)
 function product_distribution(dists::V) where {S<:ValueSupport,T<:UnivariateDistribution{S},V<:AbstractVector{T}}
     return Product{S,T,V}(dists)
 end
+
+#### Fitting
+
+"""
+    fit_mle(g::Product, x::AbstractMatrix)
+    fit_mle(g::Product, x::AbstractMatrix, γ::AbstractVector)
+
+The `fit_mle` for a multivariate Product distributions `g` is the `product_distribution` of `fit_mle` of each components of `g`.
+"""
+function fit_mle(g::Product, x::AbstractMatrix, args...)
+    d = size(x, 1)
+    length(g) == d || throw(DimensionMismatch("The dimensions of g and x are inconsistent."))
+    return product_distribution([fit_mle(g.v[s], y, args...) for (s, y) in enumerate(eachrow(x))])
+end
```

Comment:
My bad I did not see the new product framework. 
I believe this can be adapted to `ProductDistribution`. I'll do that.

Reply:
For now, I keep `Product` because it seems that `d = product_distribution([Exponential(4), Normal(1)])` `isa Product` but not a `Distributions.ArrayOfUnivariateDistribution`.
I guess when `Product` are removed, `d` will be an `DistributionsVectorOfUnivariateDistribution` and the code I added will work.